packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -58,7 +58,7 @@ export class SummarizerNode implements IRootSummarizerNode {
     }
 
     protected readonly children = new Map<string, SummarizerNode>();
-    private readonly pendingSummaries = new Map<string, SummaryNode>();
+    protected readonly pendingSummaries = new Map<string, SummaryNode>();
```

Comment:
It looks like this was made `protected`, but then not overridden in `SummarizerNodeWithGc`.

It can be overridden to avoid the `as SummaryNodeWithGC` cast of the values, but it provides limited type-safety since the setting could occur in the base class violating that type. I do think that's fine since we are sort of assuming a lot about the base class from within the extending class, and less in the other direction.

Reply:
It was overridden because if its private, it cannot be accessed in derived classes. I don't see any real benefit in overriding it. 
I think its fine for the  derived classes can do the cast/

Reply:
Oh okay that makes sense thanks for pointing that out.